testbed/correctness/metrics/metrics_test_harness.go

```diff
@@ -0,0 +1,110 @@
+// Copyright The OpenTelemetry Authors
```

Comment:
This seems like functionality that should be distributed between existing testbed files. For example, compare function should be part of validator.